icu4c/source/common/locid.cpp

```diff
@@ -632,6 +659,143 @@ Locale& Locale::init(const char* localeID, UBool canonicalize)
             break;
         }
 
+        if (canonicalize) {
+            UErrorCode status = U_ZERO_ERROR;
+            LocalUResourceBundlePointer metadata(ures_openDirect(NULL,"metadata",&status));
+            LocalUResourceBundlePointer metadataAlias(ures_getByKey(metadata.getAlias(),"alias",NULL,&status));
```

Comment:
With the logic I needed here, I cannot see how would using ures_getValueWithFallback make the code nicer. Looks like if I need to use it, the code will be much longer to achieve the same effect.

Reply:
I don't think it would be any longer, but we can revisit this later. I suggest you add
// TODO: Try to use ResourceDataValue and ures_getValueWithFallback() etc.